Rubberduck.Refactorings/EncapsulateField/EncapsulateFieldRefactoring.cs

```diff
@@ -1,47 +1,79 @@
-﻿using System;
+﻿
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
 using Rubberduck.Parsing.Rewriter;
 using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
 using Rubberduck.VBEditor;
 using Rubberduck.SmartIndenter;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using Rubberduck.VBEditor.Utility;
+using Environment = System.Environment;
 
 namespace Rubberduck.Refactorings.EncapsulateField
 {
     public class EncapsulateFieldRefactoring : IRefactoring
     {
+        private readonly RubberduckParserState _state;
         private readonly IVBE _vbe;
         private readonly IIndenter _indenter;
         private readonly IRewritingManager _rewritingManager;
-        private readonly IRefactoringPresenterFactory<IEncapsulateFieldPresenter> _factory;
+        private readonly IRefactoringPresenterFactory _factory;
         private EncapsulateFieldModel _model;
 
-        public EncapsulateFieldRefactoring(IVBE vbe, IIndenter indenter, IRefactoringPresenterFactory<IEncapsulateFieldPresenter> factory, IRewritingManager rewritingManager)
+        private readonly HashSet<IModuleRewriter> _referenceRewriters = new HashSet<IModuleRewriter>();
+        
+        public EncapsulateFieldRefactoring(RubberduckParserState state, IVBE vbe, IIndenter indenter, IRefactoringPresenterFactory factory, IRewritingManager rewritingManager)
         {
+            _state = state;
             _vbe = vbe;
             _indenter = indenter;
             _factory = factory;
             _rewritingManager = rewritingManager;
         }
 
-        public void Refactor()
+        private EncapsulateFieldModel InitializeModel()
         {
-            var presenter = _factory.Create();
-            if (presenter == null)
+            var selection = _vbe.GetActiveSelection();
+
+            if (!selection.HasValue)
             {
-                return;
+                return null;
             }
 
-            _model = presenter.Show();
+            return new EncapsulateFieldModel(_state, selection.Value);
+        }
+
+        public void Refactor()
+        {
+            _model = InitializeModel();
             if (_model == null)
             {
                 return;
             }
 
-            var rewriteSession = _rewritingManager.CheckOutCodePaneSession();
-            AddProperty(rewriteSession);
-            rewriteSession.TryRewrite();
+            using (var container = DisposalActionContainer.Create(_factory.Create<IEncapsulateFieldPresenter, EncapsulateFieldModel>(_model), p => _factory.Release(p)))
```

Comment:
This smells.  The `DisposalActionContainer` should be used for _non-owned_ objects that don't implement `IDisposable` and require cleanup after use.  Anything that _**we**_ own that needs disposal should be implementing `IDisposable`. This mechanism bypasses all of the built in code analysis checking that we do for non-disposal.

I'm not going to hunt all of these down, but there are other instances in the PR.

Reply:
The issue is that `IRefactoring` implementations themselves aren't `IDisposable`. When using a CW automagic factory, we need to take care to release the instances. I suppose I could have had written a `try/finally` block to ensure that the `Release` is called on the factory, but that feels like trading in 4 quarters for 1 dollar. Open to suggestion for less smelly alternatives.